scripts/linters/js_ts_linter.py

```diff
@@ -238,570 +222,459 @@ def _get_compiled_ts_filepath(self, filepath):
 
+        Returns:
+            OutputStream. An OutputStream object to retrieve the status of a
+            lint check.
+        """
         http_client_pattern = r':\n? *HttpClient'
 
         excluded_files = [
             'core/templates/services/request-interceptor.service.spec.ts'
         ]
 
         summary_messages = []
+        name = 'HTTP request'
 
-        with linter_utils.redirect_stdout(sys.stdout):
-            failed = False
-
-            for file_path in self.all_filepaths:
-                if file_path in excluded_files:
-                    continue
+        failed = False
 
-                if file_path.endswith('backend-api.service.ts'):
-                    continue
+        for file_path in self.all_filepaths:
+            if file_path in excluded_files:
+                continue
 
-                file_content = self.file_cache.read(file_path)
+            if file_path.endswith('backend-api.service.ts'):
+                continue
 
-                if re.findall(http_client_pattern, file_content):
-                    failed = True
-                    summary_message = (
-                        '%s --> An instance of HttpClient is found in this '
-                        'file. You are not allowed to create http requests '
-                        'from files that are not backend api services.' % (
-                            file_path))
-                    summary_messages.append(summary_message)
-                    python_utils.PRINT(summary_message)
-                    python_utils.PRINT('')
+            file_content = self.file_cache.read(file_path)
 
-            if failed:
-                summary_message = (
-                    '%s HTTP requests check failed' % (
-                        linter_utils.FAILED_MESSAGE_PREFIX))
-                summary_messages.append(summary_message)
-            else:
+            if re.findall(http_client_pattern, file_content):
+                failed = True
                 summary_message = (
-                    '%s HTTP requests check passed' % (
-                        linter_utils.SUCCESS_MESSAGE_PREFIX))
+                    '%s --> An instance of HttpClient is found in this '
+                    'file. You are not allowed to create http requests '
+                    'from files that are not backend api services.' % (
+                        file_path))
                 summary_messages.append(summary_message)
 
-            python_utils.PRINT(summary_message)
-            python_utils.PRINT('')
-
-        return summary_messages
+        return concurrent_task_utils.OutputStream(
+            name, failed, summary_messages, summary_messages)
 
     def _check_ts_ignore(self):
-        """Checks if ts ignore is used."""
+        """Checks if ts ignore is used.
+
+        Returns:
+            OutputStream. An OutputStream object to retrieve the status of a
+            lint check.
+        """
+        name = 'Ts ignore'
```

Comment:
Can we please follow some convention for naming the tests too? 

One way would be converting this camelcase `_check_ts_ignore` to `TS Ignore` -- start with capital when it has dash before it.

Reply:
I tried doing this and it looks very weird to have letter in caps in the middle. Example: `SUCCESS Ts Ignore checks passed`

Reply:
Oh yes, you're right thanks!